dts/bindings/ethernet/wiznet,wiz550io.yaml

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2024 Enphase Energy
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  WIZ550io standalone 10/100BASE-T Ethernet controller with SPI interface
+  and preconfigured unique MAC address.
+
+  This is a specialization of the W5500 Ethernet controller.
+
+  Example:
+    &spi3 {
+        status = "okay";
+        pinctrl-0 = <&spi3_sck_pb3 &spi3_miso_pb4 &spi3_mosi_pb5>;
+        pinctrl-names = "default";
+        cs-gpios = <&gpioa 4 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+
+        eth_wiz550io: eth_wiz550io@0 {
+            status = "okay";
+            compatible = "wiznet,wiz550io", "wiznet,w5500";
+            reg = <0x0>;
+            spi-max-frequency = <DT_FREQ_M(80)>;
+            int-gpios = <&gpioa 6 GPIO_ACTIVE_LOW>;
+            reset-gpios = <&gpioa 5 GPIO_ACTIVE_LOW>;
+            ready-gpios = <&gpiod 14 GPIO_ACTIVE_HIGH>;
+            preconfigured-mac-address;
+        };
+    };
+
+compatible: "wiznet,wiz550io"
```

Comment:
What's the technical reason to introduce another compat with yaml here? IMO `dts/bindings/ethernet/wiznet,w5500.yaml` is the same with additional property which can is optional.

Reply:
According to the commit message, this PR is adding support for a breakout board with a w5500 chip on it, which is not the same thing as the w5500 chip itself. The breakout board also has an MCU which has extra signals and configurations than the w5500 has on its own. I think it makes sense in this case from the perspective of DT to therefore define a new compatible.

Ill say that someone could make the argument that the new compatible should actually be a parent with a child w5500 node, but for the functionality the PR is appearing to add, I think, while accurate, that would be total overkill.

Reply:
NACK! Underlying IC is the same, which is W5500. So no additional bindings file required here IMHO. For the breakout board, it can potentially be a shield as like `boards/shields/arceli_eth_w5500/`

Reply:
I agree with @decsny, I think having a new compat makes sense.
It is more logical to a user to just specify "wiznet,wiz550io" and have the correct defaults applied (like hardware reset delay).
It also avoids polluting "wiznet,w5500" with WIZ550io specific configuration.

> Ill say that someone could make the argument that the new compatible should actually be a parent with a child w5500 node, but for the functionality the PR is appearing to add, I think, while accurate, that would be total overkill.

I took inspiration from "nxp,flexcan-fd" which is a specialization of "nxp,flexcan":
https://github.com/zephyrproject-rtos/zephyr/blob/72a17b6d887ef20b3cc08a25d8ea1d622792d57a/dts/bindings/can/nxp%2Cflexcan-fd.yaml#L12
I think this fits well as it is not really a parent (like an SPI controller to an SPI device), it's just an extension/specialization of "wiznet,w5500"

Reply:
Not a CAN expert, but AFAIK CAN-FD and CAN are two different things underlying IP blocks. W.r.t w5500, WIZ550io = W5500 + PIC12F519 with 2 additional capability to W5500,
1. Unique Hardware MAC address
2. Ready signal

So this needs to be in the same w5500 bindings with only `ready-gpio`. Regarding the MAC address, we have 2 options already,
1. local-mac-address
2. zephyr,random-mac-address
when neither presents, this eventually means hardware unique MAC.

Reply:
```
git diff dts/
diff --git a/dts/bindings/ethernet/wiznet,w5500.yaml b/dts/bindings/ethernet/wiznet,w5500.yaml
index 1490f0763b9..a634e85c9a9 100644
--- a/dts/bindings/ethernet/wiznet,w5500.yaml
+++ b/dts/bindings/ethernet/wiznet,w5500.yaml
@@ -23,3 +23,10 @@ properties:
       The reset pin of W5500 is active low.
       If connected directly the MCU pin should be configured
       as active low.
+  ready-gpios:
+    type: phandle-array
+    description: Ready pin.
+
+      The WIZ550io module has an extra GPIO pin that signals
+      when the PIC12F519 has finished configuring the W5500.
+      This is optional to shorten the hardware reset delay.
```
handle the ready-gpio as an optional property. And create a separate shield for `WIZ550io` under `boards/shields`

Reply:
Regarding the MAC address part, I agree that there isn't any reference. DTS bindings should be specific, so I propose to discuss this hardware unique address changes under `dts/bindings/ethernet/ethernet-controller.yaml`.

Reply:
> Regarding the MAC address part, I agree that there isn't any reference. DTS bindings should be specific, so I propose to discuss this hardware unique address changes under dts/bindings/ethernet/ethernet-controller.yaml.

Yes, I also don't like the ambiguity, but I will defer to the author on their opinion of if this is scope creep to discuss on this PR.

As for the shield idea, I would also like to hear @gramsay0 's opinon of this. It's not a wrong idea in general, but I will say I don't think the ready-gpio property can go on the w5500 binding if it is not part of the actual w5500 chip.

Reply:
> Not a CAN expert, but AFAIK CAN-FD and CAN are two different things underlying IP blocks. W.r.t w5500, WIZ550io = W5500 + PIC12F519 with 2 additional capability to W5500,

I don't understand the distinction you are making?

Yes, a CAN-FD controller is a different hardware IP block, it provides classical CAN + CAN-FD extensions.
WIZ550io is a hardware module, it provides w5500 + PIC12F519 extensions.

In hardware terms: WIZ550io != w5500.

> So this needs to be in the same w5500 bindings with only `ready-gpio`. Regarding the MAC address, we...

I like the idea of using the unique MAC address is neither local nor random options are enabled.
I can see that being done elsewhere already:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/ethernet/eth_esp32.c#L150

Not having a configurable `hw-reset-delay` would mean `reset-gpio`/`ready-gpio` would be required for WIZ550io, which seems ok.

> Yes, I also don't like the ambiguity, but I will defer to the author on their opinion of if this is scope creep to discuss on this PR.

I could add a comment to ethernet-controller.yaml if requested, it does seem a little out of scope.

> As for the shield idea, I would also like to hear @gramsay0 's opinon of this

I am not very familiar with shields, they don't seem very portable with things like `arduino_spi`, `mikrobus_spi`, `boosterpack_spi`, `pico_spi` hard coded..

> I will say I don't think the ready-gpio property can go on the w5500 binding if it is not part of the actual w5500 chip.

This is an excellent point.